pkg/cloudprovider/instance.go

```diff
@@ -86,11 +86,20 @@ func (p *InstanceProvider) Create(ctx context.Context, nodeTemplate *v1alpha1.AW
 		instanceTypes = instanceTypes[0:MaxInstanceTypes]
 	}
 
-	id, err := p.launchInstance(ctx, nodeTemplate, machine, instanceTypes)
+	// Get subnets given the constraints
+	subnets, err := p.subnetProvider.List(ctx, nodeTemplate)
+	if err != nil {
+		return nil, fmt.Errorf("getting subnets, %w", err)
+	}
+	if len(subnets) == 0 {
+		return nil, fmt.Errorf("no subnets matched selector %v", nodeTemplate.Spec.SubnetSelector)
+	}
+
+	id, err := p.launchInstance(ctx, nodeTemplate, machine, subnets, instanceTypes)
 	if awserrors.IsLaunchTemplateNotFound(err) {
```

Comment:
Why isn't this encapsulated in `p.launchInstance(` 🤔

Reply:
you're right, from a previous implementation, forgot to move it back. fixed